src/lib/ProductExporter.ts

```diff
@@ -133,8 +138,14 @@ export class ProductExporter {
            options.filter(option => option['enxCPQ__Parent_Product__r'] && option['enxCPQ__Parent_Product__r'][techId]===defTechId)
                   .forEach(option=> {product.options.push(option)}); 
             
+            if(this.isRelated){
+                let chargeReferenceIds = this.retrieveReferenceChargesIds(chargesIds);
+                chargeReferenceIds.forEach(chargeId =>  {product.chargesIds.push(chargeId)})
+           };
+          
            chargesIds.filter(chargeId => chargeId['enxCPQ__Root_Product__r'] && chargeId['enxCPQ__Root_Product__r'][techId]===defTechId)
-                     .forEach(chargeId =>  {delete chargeId['enxCPQ__Root_Product__r']
+                     .forEach(chargeId =>  {delete chargeId['enxCPQ__Root_Product__r'];
```

Comment:
I think it would be clearer to use map() instead of forEach() with push(). But I see that it is actually not your current change, so don't feel forced to alter that my way.